pkg/proto/aksnodeconfig/v1/tlsbootstrappingconfig.proto

```diff
@@ -1,13 +1,26 @@
 syntax = "proto3";
 package aksnodeconfig.v1;
- 
+
+enum BootstrappingAuthMethod  {
```

Comment:
It's a good practice to have zero option as `unspecified` in protobuf enums. (We probably should have a linter for protobuf?)

Otherwise, it will always default to the first option, which frequently is unexpected behaviour.

Reply:
It's ok to default to bootstrap token. But it should be explicitly, rather than a side effect of JSON unmarshalling.

Reply:
Also, other enums use different style

Reply:
https://www.sohamkamani.com/golang/enums/#default-enum-values

which explain the usage of `iota` as well as the `default` value `0` used with an undefined value

Reply:
Leaving as an int as this is consistent with the other enums in the protobuf specification.